src/main/scala/chisel3/util/circt/Synthesis.scala

```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3.util.circt
+
+import chisel3._
+import chisel3.experimental.IntrinsicModule
+import chisel3.internal.Builder
+
+import circt.Intrinsic
+
+/** A 2-to-1 mux cell intrinsic.
+  */
+private class MuxCell2Intrinsic[T <: Data](gen: T) extends IntrinsicModule("circt_muxcell2") {
+  val sel = IO(Input(UInt(1.W)))
+  val high = IO(Input(gen))
+  val low = IO(Input(gen))
+  val out = IO(Output(gen))
+}
+
+object MuxCell2 {
+
+  /** Creates an intrinsic which is lowered to a 2-to-1 MUX cell in synthesis tools.
+    * @example {{{
+    * v := MuxCell2(sel, high, low)
```

Comment:
It might be interesting to include the Chisel code that would simulate this cell in the documentation. This way a user who knows Chisel could immediately understand the exact behavior.
In this case you could just say that `MuxCell2(a,b,c) === Mux(a,b,c)` for all `a,b,c`.